pkg/controller/controlplane/valuesprovider.go

```diff
@@ -503,6 +503,34 @@ func getControlPlaneShootChartValues(
 	}, nil
 }
 
+// getStorageClassChartValues collects and returns the shoot storage-class chart values.
+func (vp *valuesProvider) GetStorageClassesChartValues(
+	_ context.Context,
+	cp *extensionsv1alpha1.ControlPlane,
+	_ *extensionscontroller.Cluster,
+) (map[string]interface{}, error) {
+	managedDefaultStorageClass := true
+	managedDefaultVolumeSnapshotClass := true
+
+	// Decode providerConfig
+	cpConfig := &apisgcp.ControlPlaneConfig{}
+	if cp.Spec.ProviderConfig != nil {
+		if _, _, err := vp.decoder.Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
+			return nil, fmt.Errorf("could not decode providerConfig of controlplane '%s': %w", kutil.ObjectName(cp), err)
+		}
+	}
+
+	if cpConfig.Storage != nil && cpConfig.Storage.ManagedDefaultStorageClass != nil && cpConfig.Storage.ManagedDefaultVolumeSnapshotClass != nil {
```

Comment:
This is not an entirely correct assertion. You could for example not specify any value for the managed snapshot class in your configuration (and hence have it nil) but want to disable the default storageclass for example.

If you feel like the default storage class and snapshot class must be configured "together" and only like that, you should change the API spec to only expose one boolean instead